libraries/standard/http/src/http_client.c

```diff
@@ -196,10 +196,102 @@ HTTPStatus_t HTTPClient_AddHeader( HTTPRequestHeaders_t * pRequestHeaders,
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( ( rangeStartOrlastNbytes < 0 ) &&
+             ( rangeEnd != HTTP_RANGE_REQUEST_END_OF_FILE ) )
+    {
+        IotLogErrorWithArgs( "Parameter check failed: Invalid range values: "
+                             "rangeEnd should be -1 when rangeStart < 0: "
+                             "RangeStart=%d, RangeEnd=%d",
+                             rangeStartOrlastNbytes, rangeEnd );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( ( rangeEnd != HTTP_RANGE_REQUEST_END_OF_FILE ) &&
+             ( rangeStartOrlastNbytes > rangeEnd ) )
+    {
+        IotLogErrorWithArgs( "Parameter check failed: Invalid range values: "
+                             "rangeStart should be < rangeEnd when both are >= 0: "
+                             "RangeStart=%d, RangeEnd=%d",
+                             rangeStartOrlastNbytes, rangeEnd );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else
+    {
+        /* Populate the buffer with the value data for the Range Request.*/
+        stdRetVal = snprintf( rangeValueBuffer,
+                              sizeof( rangeValueBuffer ),
+                              "%s%d",
+                              RANGE_REQUEST_HEADER_VALUE_PREFIX,
+                              rangeStartOrlastNbytes );
+        assert( ( stdRetVal >= 0 ) &&
+                stdRetVal <= ( int ) ( RANGE_REQUEST_HEADER_VALUE_PREFIX_LEN + MAX_INT32_NO_OF_DIGITS ) );
+        rangeValueLength += ( size_t ) stdRetVal;
+
+        /* Add remaining value data depending on the range specification type. */
+
+        /* Add rangeEnd value if request is for [rangeStart, rangeEnd] byte range */
+        if( rangeEnd != HTTP_RANGE_REQUEST_END_OF_FILE )
+        {
+            /* Add the rangeEnd value to the request range .*/
+            stdRetVal = snprintf( rangeValueBuffer + rangeValueLength,
+                                  sizeof( rangeValueBuffer ) - rangeValueLength,
+                                  "%s%d",
+                                  DASH_CHARACTER,
+                                  rangeEnd );
+            assert( ( stdRetVal >= 0 ) &&
+                    stdRetVal <= ( DASH_CHARACTER_LEN + MAX_INT32_NO_OF_DIGITS ) );
+            rangeValueLength += ( size_t ) stdRetVal;
+        }
+        /* Case when request is for bytes in the range [rangeStart, ). */
+        else if( rangeStartOrlastNbytes >= 0 )
+        {
+            /* Add the "-" character.*/
+            stdRetVal = snprintf( rangeValueBuffer + rangeValueLength,
+                                  sizeof( rangeValueBuffer ) - rangeValueLength,
+                                  "%s",
+                                  DASH_CHARACTER );
+            /* Check that only a single character was written. */
+            assert( stdRetVal == DASH_CHARACTER_LEN );
+            rangeValueLength += ( size_t ) stdRetVal;
+        }
+        else
```

Comment:
Check the complexity on this function, I believe it is probably > 8.

Reply:
I ran complexity locally, and it yielded a threshold score for the function
![image](https://user-images.githubusercontent.com/5158611/80829397-e5cf9600-8b9b-11ea-9b59-2cd77d87be0e.png)